sdk/core/AzureCore/Source/Pipeline/Policies/ContentDecodePolicy.swift

```diff
@@ -194,6 +187,7 @@ public class ContentDecodePolicy: PipelineStage {
 
         // Store the logger so that the XML parser delegate functions can access it
         logger = response.logger
+        xmlParser?.logger = logger
```

Comment:
Also, since not every request will need an xmlParser, maybe it would be good to make the property a `lazy` property, and then only assign the logger from within the `parse` method (by which time we know we need to have an xmlParser available).

Reply:
Done!